### PR TITLE
feat(calendar): spark-413846 replace id with uri when call calendar api getPartipants and getNotes

### PR DESCRIPTION
**Pull request by JudyZhuHz**
# COMPLETES #< INSERT LINK TO ISSUE >
https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-413846

## This pull request addresses

< DESCRIBE THE CONTEXT OF THE ISSUE >
spark-413846 replace id with uri when call calendar api getPartipants and getNotesByUrl

< DESCRIBE YOUR CHANGES >
1. add new function: getNotesByUri
2. getParticipants replace id with uri  

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/internal-plugin-calendar/src/calendar.js [hunk 1]
```diff
@@ -245,15 +245,26 @@ const Calendar = WebexPlugin.extend({
...
     return this.request({
       method: 'GET',
-      service: 'calendar',
-      resource: `calendarEvents/${btoa(id)}/participants`,
+      uri: participantsUrl,
+    });
+  },
+
+  /**
+   * Retrieves a collection of meetings based on the request parameters
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-calendar/src/calendar.js:260]: this comment doesn't seem to match what the function is doing

File: packages/@webex/internal-plugin-calendar/src/calendar.js [hunk 2]
```diff
@@ -294,7 +305,7 @@ const Calendar = WebexPlugin.extend({
         meetingObjects.forEach((meeting) => {
           if (!meeting.encryptedParticipants) {
             promises.push(
-              this.getParticipants(meeting.id).then((notesResponse) => {
+              this.getParticipants(meeting.participantsUrl).then((notesResponse) => {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-calendar/src/calendar.js:308]: is it guaranteed that we always have `meeting.participantsUrl` at the point in time when this code is executed?

File: packages/@webex/internal-plugin-calendar/src/calendar.js [hunk 3]
```diff
@@ -245,15 +245,26 @@ const Calendar = WebexPlugin.extend({
...
 
   /**
-   * Retrieves an array of meeting participants for the meeting id
-   * @param {String} id
+   * Retrieves an array of meeting participants for the meeting participantsUrl
+   * @param {String} participantsUrl
    * @returns {Promise} Resolves with an object of meeting participants
    */
-  getParticipants(id) {
+  getParticipants(participantsUrl) {
```
Comment (marcin-bazyl) [inline packages/@webex/internal-plugin-calendar/src/calendar.js:252]: I'm beginning to wonder what's the point of having these functions like getParticipants() and getNotesByUrl() in this internal-plugin-calendar. They don't do anything, they don't have any logic, just send a GET request and the caller has to know which special URL to pass to them so they seem a bit useless. Why do we have them? Are they going to be expanded later with more logic and this is just a start? Where are they called from?

Comment (JudyZhuHz) [reply packages/@webex/internal-plugin-calendar/src/calendar.js:252]: I checked code, when list  if the Calendar API doesn't send list of participants with the meeting object, need send getParticipants call for get meeting participants, i think happens for large meeting.

File: packages/@webex/internal-plugin-calendar/src/calendar.js (hunk 1 above)
Comment (JudyZhuHz) [reply packages/@webex/internal-plugin-calendar/src/calendar.js:260]: copy from getNotes function. Maybe i changed to get meeting notes using notesUrl from meeting object.

File: packages/@webex/internal-plugin-calendar/src/calendar.js (hunk 3 above)
Comment (JudyZhuHz) [reply packages/@webex/internal-plugin-calendar/src/calendar.js:252]: From any space, click the Meetings tab (Messages, People, Content, Meetings) and the Calendar and Meeting content tabs should call getNotest.

Comment (JudyZhuHz) [reply packages/@webex/internal-plugin-calendar/src/calendar.js:252]: As origin design, call syncCalendar, return meeting object, if doesn't send meeting notes with the meeting object, need send getNotes call for get meeting notes, remove it in 2023.5.16 (PR: https://github.com/webex/webex-js-sdk/pull/2805)

**Pull request merged**